airbyte-integrations/connectors/source-google-ads/unit_tests/common.py

```diff
@@ -38,9 +40,22 @@ def search(self, search_request):
         raise GoogleAdsException("This mocked class always returns an error")
 
 
+def make_google_ads_exception(failure_code: int):
+    # There is no easy way I could find to mock a GoogleAdsException without doing something heinous like this
+    # Following the definition of the object here
+    # https://developers.google.com/google-ads/api/reference/rpc/v10/GoogleAdsFailure
+    protobuf_as_json = json.dumps({"errors": [{"error_code": {"request_error": failure_code}, "message": "it failed"}], "request_id": "1"})
+    failure = type(GoogleAdsFailure).from_json(GoogleAdsFailure, protobuf_as_json)
+    return GoogleAdsException(None, None, failure, 1)
+
+
 class MockErroringGoogleAdsClient:
-    def __init__(self, config):
-        self.config = config
+    def __init__(self, credentials, **kwargs):
+        self.config = credentials
+        self.customer_ids = ["1"]
```

Comment:
this was needed because various pieces of code call `.customer_ids` on the client object